src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs

```diff
@@ -984,6 +984,33 @@ public class C
                 GetCSharpResultAt(32, 17, DeclarePublicApiAnalyzer.OverloadWithOptionalParametersShouldHaveMostParameters, "Method6", DeclarePublicApiAnalyzer.OverloadWithOptionalParametersShouldHaveMostParameters.HelpLinkUri));
         }
 
+        [Fact, WorkItem(4766, "https://github.com/dotnet/roslyn-analyzers/issues/4766")]
+        public async Task TestObsoleteOverloadWithOptionalParameters_NoDiagnostic()
+        {
+            var source = @"
+using System;
+
+public class C
+{
+    public void M(int p1 = 0) { }
+
+    [Obsolete]
+    public void M(char p1, int p2) { }
```

Comment:
Could you add an overload to test RS0026 too?

Reply:
@genlu `RS0026` is currently not excluded. There is an open question from @Youssef1313 on this.

Reply:
@Evangelink I excluded it in 3944643 based on https://github.com/dotnet/roslyn-analyzers/pull/4773#issuecomment-772635499.

This only needs a test. Will add it soon.

Reply:
My bad, sorry :)